### ISSUE TITLE
Unified NativeProcess calls for command line

### ISSUE THREAD

**Issue by piotrzarzycki21** [prominic/Moonshine-IDE#518]
We have huge amount of places where we are calling NativeProcess to use command line in the system. It's need to be unified at some point so we could have one very basic base class and several others which we create and override methods if it's really needed.

**Comment by rat-moonshine** [comment]
I have already a process in place for executing queued NativeProcess running in different memory thread (using Worker) working for Git processes. There you can subscribe for result return (standardError, standardOutput etc.), do different tasks based on given _type_, you can also run multiple queued grouped processes at a time.

**Comment by piotrzarzycki21** [comment]
@rat-moonshine Can you point me on classes and implementation of that whole process ?

**Comment by rat-moonshine** [comment]
Yes, sure thing Piotr. I shall prepare some documentation, I need to arrange some time for that.

**Comment by piotrzarzycki21** [comment]
It doesn't have to be anything fancy. I just need some pointers here and understand how your code and classes can be used in other areas, for example for Maven build.
